utils/shared.py

```diff
@@ -35,6 +36,26 @@ def get_r2dt_version_header():
     return header
 
 
+def cmfetch(model_id: str, cm_library="") -> str:
+    """Use cmfetch to get the covariance model."""
+    cm_temp_dir = Path(tempfile.gettempdir()) / "cms"
+    cm_file = cm_temp_dir / f"{model_id}.cm"
+    if cm_file.exists():
+        return cm_file
+    cm_temp_dir.mkdir(parents=True, exist_ok=True)
+    if not cm_library:
```

Comment:
super minor: it's often considered a slightly better practice to avoid the `not` in `if` statements with 2 branches, because reversing the if statement typically makes more sense in terms of the logic flow.

```python
if cm_library:
   # use cm_library
else:
   # use smth else
```